extensions/reading-list/src/add-page-with-tag.tsx

```diff
@@ -0,0 +1,104 @@
+import { Action, ActionPanel, Form, Icon, LocalStorage, showToast, Toast } from "@raycast/api";
+import { useState } from "react";
+import { TagForm, useTag } from "./features/tag";
+import { Page } from "./types";
+import { getMetadata } from "./utils/metadata";
+
+export default function Command() {
+  const [url, setUrl] = useState("");
+  const [selectedTagIds, setSelectedTagIds] = useState<string[]>([]);
+  const [isCreatingTag, setIsCreatingTag] = useState(false);
+  const { tags, isLoading, createTag } = useTag();
+
+  async function handleSubmit() {
+    if (!url) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "URL is required!",
+      });
+      return;
+    }
+
+    try {
+      const metadata = await getMetadata(url);
```

Comment:
When the metadata cannot be retrieved, it prevents adding the page, which is unusual. This means the user won't be able to add the private page URL.